.travis/analyse.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -e
+
+function finish() {
+  make logs stop
+}
+
+trap finish EXIT
+
+make start wait-healthy
+
+make exec command="npx hydra-validator analyse http://localhost:8080/"
```

Comment:
Wonder if this is better Dockerised?

Reply:
Trying to run it locally to check...

Reply:
Related: https://github.com/libero/article-store/pull/33/commits/53a41cc04fa63823c97e752605094f8f769ab544

Reply:
Ah, it is Dockerized, using the application (`dev`) image

Reply:
Sorry, meant Dockerise the tool itself (rather than having it as an NPM dependency).

Reply:
How would you propose building the image and then use it?

Reply:
`docker-compose` doesn't have jobs (in lieu of long-running services) so it's always a bit unnatural to include these as separate containers. We make do sometimes with a `ci` service with a `bash` command that exits with 0 during `docker-compose up`.

Reply:
```
docker-compose --file .docker/docker-compose.yml --file .docker/docker-compose.dev.yml exec app npx hydra-validator analyse htt
p://localhost:8080/
  INFO      Analysis started...  +0ms
  SUCCESS   Successfully fetched http://localhost:8080/  +0ms
  SUCCESS   -- Response status 200  +1ms
  SUCCESS   -- Api Documentation link found  +1ms
  SUCCESS   -- Successfully fetched http://localhost:8080/doc  +6ms
  SUCCESS   ---- Response status 200  +0ms
  SUCCESS   ---- Successfully parsed 55 triples  +20ms
  SUCCESS   ---- Entrypoint found: http://localhost:8080/  +1ms
  INFO      ------ Skipping already visited resource <http://localhost:8080/>  +47ms
  SUCCESS   ---- Found 2 Supported Classes  +1ms
  SUCCESS   -- Successfully parsed 3 triples  +1ms
  SUCCESS   Analysis complete Success: 9 +0ms
```

Reply:
Wouldn't want to build it locally/add it to Docker Compose, just have something like `docker run  --net=host tpluscode/hydra-validator analyse http://localhost:8080` here (or, better, in the Makefile).

Reply:
If there is an image available (or can be built elsewhere) this can make this build faster and more stable, like you would download a Selenium container rather than installing [`selenium-standalone-jar`](https://www.npmjs.com/package/selenium-standalone-jar). The base questions are whether this application needs to load the project code and/or to run in the same process, the answer seems to be no to both.